mozjs/tests/callback.rs

```diff
@@ -64,12 +64,13 @@ unsafe extern "C" fn puts(context: *mut JSContext, argc: u32, vp: *mut Value) ->
     let arg = mozjs::rust::Handle::from_raw(args.get(0));
     let js = mozjs::rust::ToString(context, arg);
     rooted!(in(context) let message_root = js);
-    EncodeStringToUTF8(context, message_root.handle().into(), |message| {
```

Comment:
Unfortunately closures won't coerce to C-ffi functions, so this becomes a bit more awkward to use.

There are currently no uses of this in servo, so no coordination is needed.

Reply:
I wonder if we can make it generic over closure `rust::EncodeStringToUTF8<F>(..., cb_c: F)` then use it like:
```rust
unsafe extern "C" fn cb(message: *const core::ffi::c_char) {
    cb_c()
}
EncodeStringToUTF8(..., cb)
```

but this is just an idea.

Reply:
I think it's not really needed (at least while we [store DOMString as rust str internally instead of utf-16](https://github.com/servo/servo/blob/37634b1251a10cdb873750456026ff3bf68693c3/components/script_bindings/str.rs#L156)). Also SMDOC says its [DEPRECATED](https://searchfox.org/mozilla-central/rev/5c3577d8865aa2d61fa89fd453d803b7c4f8b4ed/js/public/CharacterEncoding.h#410), so let's not deal with this.